python/modem-firmware-1.3+/command_interface.py

```diff
@@ -13,7 +13,7 @@
 IMEI_LEN = 15
 
 class CredentialCommandInterface(ABC):
-    def __init__(self, serial_write_line, serial_wait_for_response, verbose):
+    def __init__(self, serial_write_line, serial_wait_for_response, verbose, shell=False):
```

Comment:
This is the abstract base class for CredentialCommandInterfaces, so rather than create `shell` awareness here, you should either create that awareness in the `ATCommandInterface` subclass, or create a new subclass: `ATShellCommandInterface`

Reply:
I think the easiest solution is to expand the `ATCommandInterface` class to support shell mode.


Rather than modify the constructors, I would just have `ATCommandInterface` default to non-shell mode, and then add a method to `ATCommandInterface` named `setShellMode(shell)` to override that default.

Then call that function immediately after the constructor if we are in `AT` mode.